### DISCUSSION TITLE
function view

### DISCUSSION THREAD

**Discussion by krishno16** [Cyfrin/foundry-full-course-cu#2438] [Show and tell]
![Screenshot 2024-09-10 142441](https://github.com/user-attachments/assets/3134bc45-a715-4225-afba-5b60057c3755)

is this correct way to solve this problem

**Comment by EngrPips** [comment]
It makes sense to me.

**Reply by krishno16** [reply]
can you please tell me how can i solve this 
a view function that can be accessed from children's contracts 
is it related to scope?

**Reply by EngrPips** [reply]
That's function visibility. I am not sure I understand your question perfectly.

**Reply by krishno16** [reply]
a view function that can be accessed from children's contracts 
this is the question how can i write the code

**Reply by Anshpatwal** [reply]
> can you please tell me how can i solve this a view function that can be accessed from children's contracts is it related to scope?

Bro its kinda like Global scope and block scope in Javascript if you know, we need to declare the visibilty of the function in solidity like public, private,internal etc.

**Reply by EngrPips** [reply]
> a view function that can be accessed from children's contracts this is the question how can i write the code

```js
// SPDX-License-Identifier: GPL-3.0

pragma solidity >=0.7.0 <0.9.0;

contract Parent {
    uint256 num; 

    function setNum(uint256 _num) public returns(uint256) {
        num = _num;
        return num;
    }

    function readNum() internal view returns(uint256) {
        return num;
    }
}

contract Child is Parent {

    function getParentNum() public view returns(uint256) {
        uint256 parentNum = readNum();
        return parentNum;
    }
}
```

**Reply by krishno16** [reply]
> > can you please tell me how can i solve this a view function that can be accessed from children's contracts is it related to scope?
> 
> Bro its kinda like Global scope and block scope in Javascript if you know, we need to declare the visibilty of the function in solidity like public, private,internal etc.

should i learn javascript first before smartcontract

**Reply by EngrPips** [reply]
I think that is not necessary but if you want to it's fine and will help you more.

**Reply by Anshpatwal** [reply]
You definitely will need Javascript now , or in future cause you will need frontend to connect smart contracts . I am not an expert but right now i am learning blockchain and frontend and DSA together. I give around 1 hours to each so my brain doesnt get cooked , also you are from india right ? Nice to see fellow indians here

**Reply by EngrPips** [reply]
What if they want to go down the path of a smart contract security research.

**Reply by Anshpatwal** [reply]
> What if they want to go down the path of a smart contract security research.

Well thats a different Choice , but as Patrick said in the course that we will need Ether.js and HardHat so thats kinda connected to Javascript . @EngrPips How about you what are you Pursuing ? Blockchain Developer or something different

**Reply by EngrPips** [reply]
I am currently working on the path of security.
